candle-examples/examples/voxtral/audio_utils.rs

```diff
@@ -0,0 +1,39 @@
+/// Simple linear interpolation resampling
+/// For production use, consider using a proper resampling library
+#[must_use]
+#[allow(
+    clippy::cast_possible_truncation,
+    clippy::cast_precision_loss,
+    clippy::cast_sign_loss
+)]
+pub fn resample_audio(audio: &[f32], from_rate: u32, to_rate: u32) -> Vec<f32> {
```

Comment:
It seems snac also has a [resample audio function](https://github.com/huggingface/candle/blob/42bd33e0a6c114838e623e5e11fb466628888bbf/candle-examples/examples/snac/audio_io.rs#L248). If compatible, could you use the existing one instead?